### DISCUSSION TITLE
What is the full meta.json spec, and is there a root config file?

### DISCUSSION THREAD

**Discussion by dtbuchholz** [fuma-nama/fumadocs#413] [General]
Docs frameworks like Nextra and Docusaurus have a bit better documentation around what all of the possible meta.json fields are. Also, both have a root config file that lets you set up SEO, plugins, organize sidebars, navbars, etc. Does Fumadocs support root config like that?

**Comment by fuma-nama** [comment]
By design, we suppose these should be configured from layouts, it makes it easier to pass unserialisable data and it is my opinion on the framework. Also for SEO info, It's absolutely better to use Metadata API directly, flexible and simple.

Docs for page tree conventions exist, it introduced all APIs you could use. Page tree is only dedicated in navigation links, and I prefer making it live outside of code (this helps non-dev writers). Meta files and content should be configured without code, while leaving the maximised flexibility.

**Reply by fuma-nama** [reply]
In addition, you can read our philosophy from the Comparison section.

**Reply by dtbuchholz** [reply]
thanks @fuma-nama, appreciate the quick response! i think i'm getting the hang of it...for reference, i was previously using docusaurus.

one follow up question—is there a straightforward way to "eject" and customize existing components? for example, i want to create a custom version of the default sidebar. instead of rewriting the DocsLayout etc, it'd be nice if i could eject the sidebar, alter it, and then it gets used automatically.

**Reply by fuma-nama** [reply]
If the exposed options (like banner and footer) cannot fulfil your needs, you will have to replace it with your own component. 

As Fumadocs is open source, you can clone the source code of sidebar from our repo directly. Note: you can pass server components!

**Reply by fuma-nama** [reply]
I assume you have read https://fumadocs.vercel.app/docs/ui/blocks/layout#sidebar

**Reply by dtbuchholz** [reply]
Gotcha, that makes sense...i'll take a stab at that approach. And I actually _didn't_ see that docs page...somehow missed it. My fault!

Btw, I think this describes the idea I was looking for, which is planned in Nextra v4: https://github.com/shuding/nextra/issues/332

**Reply by fuma-nama** [reply]
And with App router, we can actually send *only* the used layout to client bundle, this is pretty attractive to me.
